calblueprint/dccentralkitchen
# Link store address to directions in maps app

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
* Tap the address line of a store to either:
  * Get directions in an external maps app
  * Copy the address to clipboard

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links

### Online sources
[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"
n/a
### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
n/a
## How to review

[//]: # "The order in which to review files and what to expect when testing locally"
* Make sure to set your simulator location to be near the DC area (otherwise it might open similarly-named stores...not sure how to fix this)
![image](https://user-images.githubusercontent.com/21160510/77599501-19930e00-6ec2-11ea-8578-bc37f2c16f47.png)

* Try opening directions for a bunch of different stores—it searches for the nearest store by address closet to the store's longitude, there may be some exceptions where it doesn't find the right store. I ended up abandoning longitude/latitude bc it was really jank for Android, so I went with a query string that may be hit or miss.
## Next steps
[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

* Look into [Action Sheets](https://github.com/expo/react-native-action-sheet) instead of Alerts 
## Tests Performed, Edge Cases
[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

* Tried to simulate cross-platform compatibility by setting it to Google maps for Android 
### Screenshots
https://www.loom.com/share/836be07d58664c77a39bdd62dfdc05de
![image](https://user-images.githubusercontent.com/21160510/77599408-c9b44700-6ec1-11ea-9d03-cdfba3159f92.png)
![image](https://user-images.githubusercontent.com/21160510/77599413-cf119180-6ec1-11ea-9c0f-6285f19c0b3c.png)


[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."
